pkg/epp/plugins/plugins.go

```diff
@@ -25,4 +25,21 @@ type Plugin interface {
 
 // Handle provides plugins  set of standard data and tools to work with
 type Handle interface {
+	// Plugins returns the sub-handle for working with instantiated plugins
+	Plugins() HandlePlugins
+}
```

Comment:
I'm not sure I understand why we have here interface inside interface.

Reply:
The idea behind the plugins.Handle interface was to provide EPP information to plugins.

My assumption is that there will be many types of information.. As such I wanted to group the plugin stuff together. In general the plugins "sub-handle" is to allow plugin factories to know about other plugin instances. This is needed for the DecisionTree filter.

Reply:
what's the downside of having these functions under Handle interface? (instead of `Plugins()):
```go
// Plugin returns the named plugin instance
	Plugin(name string) Plugin

	// AddPlugin adds a plugin to the set of known plugin instances
	AddPlugin(name string, plugin Plugin)

	// GetAllPlugins returns all of the known plugins
	GetAllPlugins() []Plugin

	// GetAllPluginsWithNames returns all of the known plugins with their names
	GetAllPluginsWithNames() map[string]Plugin
``` 

is there any benefit for separating into additional interface?

Reply:
@ahg-g Who originally proposed this interface spoke about APIs for the queues for suport with the upcoming queueing layer. As time proceeds there will probably be more and more APIs here. Having each "type" of APIs separate will keep the overall API cleaner.

See https://pkg.go.dev/k8s.io/kubernetes/pkg/scheduler/framework#Handle for an example